drivers/flash/flash_stm32_qspi.c

```diff
@@ -65,6 +65,86 @@ LOG_MODULE_REGISTER(flash_stm32_qspi, CONFIG_FLASH_LOG_LEVEL);
 /* In dual-flash mode, total size is twice the size of one flash component */
 #define STM32_QSPI_DOUBLE_FLASH	DT_PROP(DT_NODELABEL(quadspi), dual_flash)
 
+/*
+ * A register of the flash device, such as a status register.
+ *
+ * When dual-flash mode is enabled, this structure contains the value of the actual register of both
+ * flash memories. For example, if an instance of this structure is used to hold the value of the
+ * status register, 'flash0_val' and 'flash1_val' will be equal respectively to the value of the
+ * status register of the first and second flash memory.
+ *
+ * This structure is packed as it is directly sent/received over the QSPI bus.
+ */
+struct flash_reg {
+	uint8_t flash0_val;
+#if STM32_QSPI_DOUBLE_FLASH
+	uint8_t flash1_val;
+#endif /* STM32_QSPI_DOUBLE_FLASH */
+} __packed;
+
+/*
+ * Sets a bit in a flash register.
+ *
+ * In dual-flash mode, the value is updated for both flash memories.
+ */
+static inline void flash_reg_set_for_all(struct flash_reg *reg, uint8_t bit)
+{
+	reg->flash0_val |= bit;
+#if STM32_QSPI_DOUBLE_FLASH
+	reg->flash1_val |= bit;
+#endif /* STM32_QSPI_DOUBLE_FLASH */
+}
+
+/*
+ * Checks if a bit is set in a flash register.
+ *
+ * In dual-flash mode, this routine returns true if and only if the bit is set for both flash
+ * memories.
+ */
+static inline bool flash_reg_is_set_for_all(struct flash_reg *reg, uint8_t bit)
+{
+	bool is_set = reg->flash0_val & bit;
+
+#if STM32_QSPI_DOUBLE_FLASH
+	is_set = is_set && (reg->flash1_val & bit);
+#endif /* STM32_QSPI_DOUBLE_FLASH */
+
+	return is_set;
+}
+
+/*
+ * Checks if a bit is clear in a flash register.
+ *
+ * In dual-flash mode, this routine returns true if and only if the bit is clear for both flash
+ * memories.
+ */
+static inline bool flash_reg_is_clear_for_all(struct flash_reg *reg, uint8_t bit)
```

Comment:
Nitpick: would suggest renaming `bit` to `bitmask`  for better accuracy. 
In `spi_nor.h`, the status register "bits" are defined as bitmasks, not as individual bit positions.

For bit checks, there is `IS_BIT_SET` in `util_macro.h`.